src/CoreGraphics/CGColorSpace.cs

```diff
@@ -342,16 +365,44 @@ public byte[] GetColorTable ()
 			CGColorSpaceGetColorTable (handle, table);
 			return table;
 		}
-			
+
 		[DllImport (Constants.CoreGraphicsLibrary)]
+		[Deprecated (PlatformName.iOS, 11, 0, message: "Use 'CreateIDCCData' instead.")]
+		[Deprecated (PlatformName.MacOSX, 10, 13, message: "Use 'CreateIDCCData' instead.")]
+		[Deprecated (PlatformName.TvOS, 11, 0, message: "Use 'CreateIDCCData' instead.")]
+		[Deprecated (PlatformName.WatchOS, 4, 0, message: "Use 'CreateIDCCData' instead.")]
 		extern static /* CGColorSpaceRef */ IntPtr CGColorSpaceCreateWithICCProfile (/* CFDataRef */ IntPtr data);
 
+		[DllImport (Constants.CoreGraphicsLibrary)]
+		[iOS (10,0)][Mac (10,12)][Watch (3,0)][TV (10,0)]
+		extern static /* CGColorSpaceRef */ IntPtr CGColorSpaceCreateWithICCData (/* CFTypeRef cg_nullable */ IntPtr data);
+
+		[Deprecated (PlatformName.iOS, 11, 0, message: "Use 'CreateIDCCData' instead.")]
+		[Deprecated (PlatformName.MacOSX, 10, 13, message: "Use 'CreateIDCCData' instead.")]
+		[Deprecated (PlatformName.TvOS, 11, 0, message: "Use 'CreateIDCCData' instead.")]
+		[Deprecated (PlatformName.WatchOS, 4, 0, message: "Use 'CreateIDCCData' instead.")]
 		public static CGColorSpace CreateICCProfile (NSData data)
 		{
-			// older versions did not throw - so we'll return null
-			if (data == null)
-				return null;
-			var ptr = CGColorSpaceCreateWithICCProfile (data.Handle);
+			IntPtr ptr = CGColorSpaceCreateWithICCProfile (data.GetHandle ());
+			return ptr == IntPtr.Zero ? null : new CGColorSpace (ptr, true);
+		}
+
+		[Introduced (PlatformName.iOS, 10, 0)]
+		[Introduced (PlatformName.MacOSX, 10, 12)]
+		public static CGColorSpace CreateICCData (NSData data)
+		{
+			return CreateICCData (data.GetHandle ());
+		}
+
+		[iOS (10,0)][Mac (10,12)][Watch (3,0)][TV (10,0)]
+		public static CGColorSpace CreateICCData (CGDataProvider provider)
+		{
+			return CreateICCData (provider.GetHandle ());
+		}
+
+		static CGColorSpace CreateICCData (IntPtr handle)
```

Comment:
not public but let's be consistent ;)

Reply:
Fixed: e5e5cc46eb2988c7a35f7bdefe168d259b9891e2